model-optimizer/extensions/back/SpecialNodesFinalization.py

```diff
@@ -126,26 +126,24 @@ class RemoveConstToResult(BackReplacementPattern):
     def pattern():
         return dict(
             nodes=[
-                ('const_node', {'type': 'Const', 'kind': 'op'}),
-                ('const_data', {'kind': 'data'}),
+                ('const_data', {'kind': 'data', 'value': lambda value: value is not None}),
```

Comment:
Please, update the description of the transformation. It should clearly describes the case of ShapeOf sub-graph.